File: examples/v2/template-sources/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
template_suffix = config.get("templateSuffix") or "dev"
source_url = config.get("sourceUrl") or "https://github.com/pulumi/examples"

source = ps_v2.OrgTemplateCollection(
    "source",
    org_name=service_org,
    name=f"v2-templates-{template_suffix}",
    source_url=source_url,
)

pulumi.export("collectionName", source.name)

File: examples/v2/template-sources/go/main.go
package main

import (
	v2 "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		organizationName := cfg.Get("organizationName")
		if organizationName == "" {
			organizationName = "service-provider-test-org"
		}
		templateSuffix := cfg.Get("templateSuffix")
		if templateSuffix == "" {
			templateSuffix = "dev"
		}
		sourceURL := cfg.Get("sourceUrl")
		if sourceURL == "" {
			sourceURL = "https://github.com/pulumi/examples"
		}

		source, err := v2.NewOrgTemplateCollection(ctx, "source", &v2.OrgTemplateCollectionArgs{
			OrgName:   pulumi.String(organizationName),
			Name:      pulumi.String("v2-templates-" + templateSuffix),
			SourceURL: pulumi.String(sourceURL),
		})
		if err != nil {
			return err
		}

		ctx.Export("collectionName", source.Name)
		return nil
	})
}
